deploy_nixos/main.tf

```
@@ -81,6 +81,12 @@ variable "triggers" {
   default     = {}
 }
 
+variable "arguments" {
+  type        = map(any)
```

```suggestion
  type        = any
```
According to https://www.terraform.io/language/expressions/type-constraints (and testing), `map(any)` requires that all values be of the same type. This will (for example) coerce boolean values to strings, if any of the values are strings.